app/controllers/home_controller.rb

```diff
@@ -1,17 +1,25 @@
 class HomeController < ApplicationController
+  ACTIONS = [:marital_status]
   def index
   end
 
-  def marital_status
-    @marital_status = MaritalStatus.new
-  end
+  ACTIONS.each do |action|
+    define_method(action.to_s) do
+      instance_variable_set("@#{action.to_s}", model_instance(action))
+    end
 
-  def marital_status_save
-    session[:marital_status] = params["marital_status"]["married"]
-    redirect_to :summary
+    define_method("#{action}_save") do
+      session[action] = params[action.to_s]["married"]
+      redirect_to :summary
+    end
   end
```

Comment:
What is meant to happen here:

* create a **show** action
* create a **save** action

Both from the name of the model, which is stored in **ACTION**.